astropy/table/tests/test_table.py

```diff
@@ -1088,6 +1091,32 @@ def test_convert_numpy_array(self, table_types):
         with pytest.raises(ValueError):
             np_data = np.array(d, dtype=[(str('c'), 'i8'), (str('d'), 'i8')])
 
+    def test_byteswap_fits_array(self, table_types):
```

Comment:
Ah, that's true.  I didn't really think about explicitly testing this code on big-endian too.  It shouldn't much matter, but it can't hurt to add a more generic test like this.